lib/order.rb

```diff
@@ -0,0 +1,44 @@
+require_relative 'menu'
+require_relative 'restaurant'
+
+class Order
+
+attr_reader :basket, :total
+
+def initialize(restaurant = Restaurant.new)
+  @restaurant = restaurant
+  @basket = Hash.new(0)
+  @total = 0
+end
+
+def check_menu
+  @restaurant.menu.check
+end
+
+def add_item(dish, quantity = 1)
+  chosen_dish = @restaurant.menu.select_by_name(dish)
```

Comment:
instead of reaching through `@restaurant` to get to the menu, you could initialize the object with a menu instead of a restaurant.